ecs-cli/modules/utils/compose/ecs_params_reader.go

```diff
@@ -98,6 +116,85 @@ func (cd *ContainerDef) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+// HealthCheck.UnmarshalYAML is a custom unmarshaler for healthcheck that parses
+// both the docker compose and ECS syntaxes
+func (h *HealthCheck) UnmarshalYAML(unmarshal func(interface{}) error) (err error) {
```

Comment:
I mostly agree with your approach, except that I would merge together any overrides with what is passed in via docker compose before setting it on the output HealthCheck. Mostly, my issue is having an `ecs.HealthCheck` type defined on `ECSParams` in the reader; to me, the parsing step shouldn't really know anything about ECS types from the sdk -- that's the job of the converter. In playing around with this code myself (See: https://github.com/PettitWesley/amazon-ecs-cli/pull/1/commits/071da90aaded24cd207b905cb02e88d2e13d0f6a) , I think this tells me that really the `ecs_params_reader` and `convert_task_definition` belong in different packages, since they have their own concerns.